allauth/socialaccount/providers/apple/provider.py

```diff
@@ -0,0 +1,31 @@
+from allauth.account.models import EmailAddress
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class AppleProvider(OAuth2Provider):
+    id = 'apple'
+    name = 'Apple'
+    account_class = ProviderAccount
+
```

Comment:
We should include `name, email` in `get_default_scope` method.

Reply:
Whether or not `email` is in the default scope should depend on `from allauth.socialaccount.app_settings import QUERY_EMAIL`.